lib/server.go

```diff
@@ -3297,3 +3298,13 @@ func (progress *SyncProgress) PrintLoop() {
 		}
 	}
 }
+
+type NetworkManagerConnections struct {
+	NonValidators    []*RemoteNode
+	Validators       *collections.ConcurrentMap[bls.SerializedPublicKey, *RemoteNode]
+	ActiveValidators *collections.ConcurrentMap[bls.SerializedPublicKey, consensus.Validator]
+}
+
+func (srv *Server) GetNetworkManagerConnections() *collections.ConcurrentMap[RemoteNodeId, *RemoteNode] {
+	return srv.networkManager.GetAllRemoteNodes()
```

Comment:
Avoid returning the index itself so it can't be mutated
```suggestion
func (srv *Server) GetNetworkManagerConnections() []*RemoteNode {
	return srv.networkManager.GetAllRemoteNodes().GetAll()
```